forge-ai/src/main/java/forge/ai/ability/EffectAi.java

```diff
@@ -383,6 +385,118 @@ protected AiAbilityDecision checkApiLogic(final Player ai, final SpellAbility sa
                     return new AiAbilityDecision(0, AiPlayDecision.CantPlayAi);
                 }
             }
+        } else if (sa.hasParam("RememberObjects")) { //generic
+            boolean cantAttack = false;
+            boolean cantBlock = false;
+            boolean cantActivate = false;
+
+            String duraction = sa.getParam("Duration");
+
+            String matchStr = "Card.IsRemembered";
+
+            for (String st : sa.getParam("StaticAbilities").split(",")) {
```

Comment:
seems good
but it really needs it's own helper function that can also be used by DetainAI and especially AttachAI because a lot of Auras are also doing that

Reply:
This is not for DetainEffect anymore, but might be useful for #8172